internal/cosesign1/create.go

```diff
@@ -0,0 +1,134 @@
+package cosesign1
+
+import (
+	"crypto/rand"
+
+	"crypto"
+	"crypto/x509"
+	"encoding/pem"
+	"io"
+	"log"
+
+	"github.com/veraison/go-cose"
+)
+
+func pem2der(chainPem []byte) []byte {
+	block, rest := pem.Decode(chainPem)
+	var r = []byte{}
+	if block.Bytes != nil {
+		r = block.Bytes
+	}
+	for len(rest) != 0 {
+		block, rest = pem.Decode(rest)
+		if block.Bytes != nil {
+			r = append(r, block.Bytes...)
+		}
+	}
+	return r
+}
+
+func CreateCoseSign1(payloadBlob []byte, issuer string, feed string, contentType string, chainPem []byte, keyPem []byte, saltType string, algo cose.Algorithm, verbose bool) (result []byte, err error) {
+	var signingKey any
+	var keyDer *pem.Block
+	keyDer, _ = pem.Decode(keyPem) // dicard remaining bytes
```

Comment:
`pem.Decode` can return `nil` block, we should return an error in that case, otherwise the line below will result in a nil pointer deref